cmd/deploy.go

```diff
@@ -472,3 +488,40 @@ func validateBuildType(buildType string) error {
 	}
 	return nil
 }
+
+func parseImageDigest(imageSplit []string, config deployConfig, cmd *cobra.Command) (deployConfig, error) {
+
+	if !strings.HasPrefix(imageSplit[1], "sha256:") {
+		return config, fmt.Errorf("value '%s' in --image has invalid prefix syntax for digest (should be 'sha256:')", config.Image)
+	}
+
+	if len(imageSplit[1][7:]) != 64 {
+		return config, fmt.Errorf("sha256 hash in '%s' from --image has the wrong length (%d), should be 64", imageSplit[1], len(imageSplit[1][7:]))
+	}
+
+	// if --build was set but not as 'disabled', return an error
+	if cmd.Flags().Changed("build") && config.BuildType != "disabled" {
+		return config, fmt.Errorf("the --build flag '%s' is not valid when using --image with digest", config.BuildType)
+	}
+
+	// if --build was not explicitly set via CLI, print a warning
+	if config.BuildType == "" {
+		fmt.Printf("Warning: --build flag not set when using --image with digest, setting to 'disabled'\n")
+	}
+
+	// if the --push flag was set by a user to 'true', return an error
+	if cmd.Flags().Changed("push") && config.Push {
+		return config, fmt.Errorf("the --push flag '%v' is not valid when using --image with digest", config.Push)
+	}
+
+	// if the --push flag was not set by a user, print a warning
+	if !cmd.Flags().Changed("push") {
+		fmt.Printf("Warning: --push flag not set when using --image with digest, setting to 'false'\n")
```

Comment:
Similar to above, I think this might be unnecessary. But this has me thinking, what happens if the image with the given digest exists only locally? Is it necessary for the user to manually push the image to the registry using `docker push`? Images built by the `func` CLI are only pushed to the registry when you run `func deploy`, so it seems perhaps we don't want to prohibit pushing the image?

Reply:
I unified the printing message and removed the conditions so it gets printed out every time image with digest is successfully parsed, does this work or were you thinking something else?

Reply:
@lance that's a great point with the push. Doesn't push change the image digest though? I'd keep the behavior as it is for now, we can always change this. This is anyway something that users probably won't use directly, but mainly through Tekton